From 16e4b0719d5b3f6d8e0cf73cfc38e3d11eb53877 Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Wed, 24 Dec 2014 18:02:13 +0100
Subject: [PATCH 0178/1173] HUE-2505 [fb] Top row is not the current directory

Updated sorting algorithm for folders that start with a special character
---
 .../filebrowser/templates/listdir_components.mako  |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/apps/filebrowser/src/filebrowser/templates/listdir_components.mako b/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
index 8b76b7e..40a9aa5 100644
--- a/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
+++ b/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
@@ -769,10 +769,16 @@ from django.utils.translation import ugettext as _
       self.isCurrentDirSentryManaged = ko.observable(false);
 
       self.filesSorting = function (l, r) {
-        if (l.name == ".." && r.name == "."){
+        if (l.name == "..") {
           return -1;
         }
-        else if (l.name == "." && r.name == ".."){
+        else if (r.name == "..") {
+          return 1;
+        }
+        else if (l.name == ".") {
+          return -1;
+        }
+        else if (r.name == ".") {
           return 1;
         }
         else {
-- 
1.7.9.5

